tribits/core/package_arch/TribitsGitRepoVersionInfo.cmake

```diff
@@ -105,26 +107,35 @@ function(tribits_git_repo_sha1  gitRepoDir  gitRepoSha1Out)
 endfunction()
 
 
-# Run the git log command to get the version info for a git repo
+# @FUNCTION: tribits_generate_commit_info_string()
 #
-function(tribits_generate_single_repo_version_string  gitRepoDir
-   repoVersionStringOut
-  )
-
-  tribits_assert_git_executable()
-
-  # A) Get the basic version info.
-
+# Get the formatted commit info containing commit's SHA1,
+# author, date, email, and 80 character summary.
+#
+# Usage:
+#   tribits_generate_commit_info_string(<gitRepoDir> <gitCommitSha1>
+#     commitInfoStringOut)
+#
+# NOTE: Below, it is fine if ${maxSummaryLen} > len(${gitCmndOutput}) as
+# string(SUBSTRING ...) will just shorten this to the length of the string.
+#
+function(tribits_generate_commit_info_string gitRepoDir gitCommitSha1
+   commitInfoStringOut
+  ) 
+  
+  # A) Get commit hash, author, date, and email
+  
   execute_process(
-    COMMAND ${GIT_EXECUTABLE} log -1 "--pretty=format:%h [%ad] <%ae>"
+    COMMAND ${GIT_EXECUTABLE} log -1 ${gitCommitSha1} "--pretty=format:%h [%ad] <%ae>"
     WORKING_DIRECTORY ${gitRepoDir}
     RESULT_VARIABLE gitCmndRtn
     OUTPUT_VARIABLE gitCmndOutput
+    OUTPUT_STRIP_TRAILING_WHITESPACE  ERROR_STRIP_TRAILING_WHITESPACE
```

Comment:
I guess we could put these args on the same line as well if they fit in 80 chars.